Source/TaskDelegate.swift

```diff
@@ -182,8 +180,12 @@ class DataTaskDelegate: TaskDelegate, URLSessionDataDelegate {
         }
     }
 
-    var dataProgress: ((_ bytesReceived: Int64, _ totalBytesReceived: Int64, _ totalBytesExpectedToReceive: Int64) -> Void)?
-    var dataStream: ((_ data: Data) -> Void)?
+    var progress: Progress
+
+    var progressHandler: (closure: Request.ProgressHandler, queue: DispatchQueue)?
+    var progressDebugHandler: (closure: Request.DownloadProgressHandler, queue: DispatchQueue)?
+
+    var dataStream: ((_ data: Data) -> Void)? // this should use handler pattern also
```

Comment:
Removed in eca9afa9. I don't think we need a custom dispatch queue for the stream closure unless someone really wants to see that. IMO it's probably best to call that directly from the session delegate queue.